File: src/main/java/bad/robot/radiate/ui/ChessBoardLayout.java
package bad.robot.radiate.ui;

import java.awt.*;
import java.util.Collection;

@Deprecated
class ChessboardLayout extends GridLayout {

    private final Collection<?> tiles;

    public ChessboardLayout(Collection<?> tiles) {
        this.tiles = tiles;
    }

    @Override
    public void layoutContainer(Container parent) {
        Tiles tiles = EmptyTrailingTiles.tiles(this.tiles.size());
        setRows(tiles.rows());
        setColumns(tiles.columns());
        super.layoutContainer(parent);
    }

}

File: src/main/scala/bad/robot/radiate/ui/ChessBoardLayout.scala
package bad.robot.radiate.ui

import java.awt.{Container, GridLayout}

class ChessboardLayout(tiles: Seq[_]) extends GridLayout {

  override def layoutContainer(parent: Container) {
    val tiles = EmptyTrailingTiles.tiles(this.tiles.size)
    setRows(tiles.rows)
    setColumns(tiles.columns)
    super.layoutContainer(parent)
  }
}